lib/ssh/src/ssh_auth.erl

```diff
@@ -178,9 +191,29 @@ publickey_msg([SigAlg, #ssh{user = User,
                                                 ?string(SigAlgStr),
                                                 ?binary(PubKeyBlob),
                                                 ?binary(SigBlob)]},
-              Ssh);
-     	_ ->
-	    {not_ok, Ssh}
+            Ssh);
+
+        {ok, {PrivKey, PubKeyBlob}} ->
+            SigAlgStr = atom_to_list(SigAlg),
+            SigData = build_sig_data(SessionId, User, Service,
+                                     PubKeyBlob, SigAlgStr),
+            Hash = ssh_transport:sha(SigAlg),
+            Sig = ssh_transport:sign(SigData, Hash, PrivKey),
+            SigBlob = list_to_binary([?string(SigAlgStr),
+                                      ?binary(Sig)]),
+
+            ssh_transport:ssh_packet(
+                #ssh_msg_userauth_request{user = User,
+                                          service = Service,
+                                          method = "publickey",
+                                          data = [?TRUE,
+                                                  ?string(SigAlgStr),
+                                                  ?binary(PubKeyBlob),
+                                                  ?binary(SigBlob)]},
+                Ssh);
```

Comment:
This branch handles the current/old behavior (the diff somehow doesn't show this properly).